src/main.rs

```diff
@@ -1,126 +1,267 @@
+
+        i = (i + 1) % frames.len();
     }
 }
 
-#[derive(Debug, Clone)]
-struct Object {
-    id: ObjectId, //TODO
-    mesh: MeshId,
-    //TODO: See if we need to have a list of material IDs here
+/// Renders a set of axis that match the orientation of the given view matrix
+fn render_axis(
+    axis_color: &mut Buffer2d<u32>,
+    view: Mat4<f32>,
+) {
+    // Only want to load this once
+    thread_local! {
+        /// This is an example for using doc comment attributes
+        static AXIS_MESHES: Vec<Mesh> = load_frame("samples/axis/axis.obj");
+    }
+
+    let axis_size = axis_color.size();
+    let projection = Mat4::perspective_rh_no(0.35*PI, (axis_size[0] as f32)/(axis_size[1] as f32), 0.01, 100.0)
+        * Mat4::<f32>::scaling_3d(0.5);
+
+    let mut depth = Buffer2d::new(axis_color.size(), 1.0);
+    AXIS_MESHES.with(|meshes| {
+        render(axis_color, &mut depth, Mat4::identity(), view, projection, meshes, 0.0)
+    });
 }
 
-#[derive(Debug, Clone, Default)] //TODO: Look up "Default trait" and "rust auto derive"
-struct Scene {
-    objects: Vec<Object>,
-    meshes: Vec<Mesh>,
-    materials: Vec<Material>, //TODO: figure out what to get from the material (at least diffuse color)
+/// Copy the entire source buffer into the given target buffer starting at the given offset
+///
+/// Unsafe because no bounds checking is performed.
+unsafe fn copy(target: &mut Buffer2d<u32>, source: &Buffer2d<u32>, (x, y): (usize, usize)) {
+    let Vec2 {x: source_width, y: source_height} = Vec2::from(source.size());
+
+    for i in 0..source_width {
+        for j in 0..source_height {
+            let value = source.get([i, j]);
+            target.set([x + i, y + j], *value);
+        }
+    }
 }
 
-fn main() -> Result<(), AppError> {
-    let file = File::open("samples/pyramid.fbx")?;
-    let reader = EventReader::new(file);
-
-    let mut scene = Scene::default();
-
-    //TODO: Lookup "rust vecdeque"
-    let mut found_objects = false;
-    // For reference of how FBX meshes are exported, see this function: https://developer.blender.org/diffusion/BA/browse/master/io_scene_fbx/export_fbx_bin.py;2532b96844c121b710e1a1973d2a5ff824ab3be4$815
-    for event_res in reader {
-        match event_res? {
-            StartFbx(_) => {},
-            EndFbx => {},
-            //TODO: Look up "rust pattern guard" to understand this "if" syntax
-            StartNode {ref name, properties} if name == "Objects" => {
-                found_objects = true;
-            },
-            StartNode {ref name, properties} if name == "Geometry" && found_objects => {
-                let mesh = Mesh::new(); //TODO: implement new()
-                // Get vertices, polygons, and edges, and create a mesh
-                scene.meshes.push(mesh);
+fn render(
+    color: &mut Buffer2d<u32>,
+    depth: &mut Buffer2d<f32>,
+    model: Mat4<f32>,
+    view: Mat4<f32>,
+    projection: Mat4<f32>,
+    meshes: &[Mesh],
+    outline_thickness: f32,
```

Comment:
can we refer to thickness as THICC-ness pls 🤣

Reply:
```rust
outline_thiccness: f32, // thicc boi
```